### DISCUSSION TITLE
Postgres container that will be run only once per test suite

### DISCUSSION THREAD

**Discussion by Krzysztof-Lempicki** [testcontainers/testcontainers-java#5601] [Q&A] [answered]
Hi,

In my project I am using SpringBoot and Spock.

All my tests transactions are never committed.
I don't need to run container per test or test class.

How to setup postgres container that will start before first test and
will be closed after last one?

**Comment by eddumelendez** [comment, marked as answer]
You can opt-in for container reuse feature.
1. Define container with `withReuse(true)`
```groovy
@Shared
public GenericContainer container = new GenericContainer(SpockTestImages.HTTPD_IMAGE)
        .withExposedPorts(80)
        .withReuse(true)
```
2.  opt-in to reuse in `~/.testcontainers.properties`
```
testcontainers.reuse.enable=true
```
3. containers should be started manually using `container.start()`. See [docs](https://www.testcontainers.org/test_framework_integration/manual_lifecycle_control/)

> **Note**
> This is an experimental feature. Containers using `withReuse(true)` won't stop after all tests are finished.

You can create a BaseITSpecification which declares the container to be reused across your tests. Take into account that the container definition should be the same to be reused.

**Reply by kiview** [reply]
Besides the excellent answer by @eddumelendez, you can also make use of the [Singleton Container Pattern](https://www.testcontainers.org/test_framework_integration/manual_lifecycle_control/#singleton-containers).
